yaba.yaml

```
@@ -0,0 +1,321 @@
+openapi: "3.0.0"
+info:
+  version: 0.0.1
+  title: BETYdb-YABA
+  description: <h2>The YABA API (BETYdb-YABA) is a Standardized RESTful Web Service API Specification for MetaData Upload to respective tables.</h2>
+  license:
+    name: BSD 3 Clause
+servers:
+  - url: http://localhost:8080/yaba/v1
+paths:
+  /experiments:
+    post:
+      summary: Upload metadata to Experiments Table
+      operationId: metaExperiments
+      tags:
+        - experiments
+      requestBody:
+        content:
+          application/json:
+            schema:      # Request body contents
+              type: object
+              properties:
+                name:
+                  type: string
+                start_date:
+                  type: string
+                end_date:
+                  type: string
+                user_id:
+                  type: integer
+              example:   # Sample data
+                name: 'MAC Season 6: Sorghum BAP'
+                start_date: '2018-04-06'
+                end_date: '2018-08-01'
+                user_id: 6000000004
+      responses:
+        '201':
+          description: Successfully Inserted to Experiment Table
+        default:
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+  /sites:
+    post:
+      summary: Upload metadata to Sites Table
+      operationId: metaSites
```

```suggestion
      operationId: metaSites
```

what does 'operationId' imply? Should there be one per table, or one per action? for example, if there is one operation to insert a new record to the sites table should it be 'insertSite'?

---

It defines the function/method which is called in response to that POST request.So,it should be one per action .We can put any name to that function.